[Wrath] Switch to `RegisterForLease`, fix various methods

zbee opened the pull request:
In this PR:
- [X] Switches to `RegisterForLease` instead of `RegisterForLeaseWithCallback`
- [X] Fixes methods that have incorrect signatures because they now return `SetResult` values
- [X] Fixes the `AutoRotationConfigOption` methods to have correct signatures
  - [X] Increases documentation for these methods
  - [X] Includes the enums for the two `Option`s that did not have their matching type in the code

Maybe another time:
- [ ] Add `RegisterForLeaseWithCallback` ?
  - [ ] Allow IPC method creation in Lua
- [ ] Include other Wrath methods

zbee committed:
fix(Wrath): +`RegisterForLease`

Switches from Wrath's `RegisterForLeaseWithCallback` to simply `RegisterForLease`, as implementation of IPC methods via lua scripts is not yet supported, so that was only going to guarantee errors.
Also wraps `RegisterForLease` to always provide SND's internal name.

zbee committed:
fix(Wrath): ~`SetResult` methods

Fixes several Wrath methods that return `SetResult` to now have the correct signature.
Includes the `SetResult` Enum.

zbee committed:
fix(Wrath): ~`AutoRotationConfigOption` methods

Fixes the `Set` and `Get` `AutoRotationConfigOption` methods to have the correct signatures.
Includes more documentation for the newly-added parameters.
Includes the appropriate enums for the two options that were missing their value types.

Jaksuhn commented:
any enums you add as return values have to be added in `IPCModule.Register` btw

zbee committed:
fix(Wrath): +Enums in `IPCModule.Register`

Include the new Wrath Enums in `Register()`, so they can actually be used.

zbee committed:
Merge remote-tracking branch 'upstream/main'

# Conflicts:
#	SomethingNeedDoing/LuaMacro/Modules/IPCModule.cs

zbee commented:
@Jaksuhn Thanks for the tip, added that.
I've taken this out of draft as I'm not going to get to those other tasks in a timely manner.

Jaksuhn commented on SomethingNeedDoing/External/Wrath.cs:
```diff
@@ -6,12 +6,15 @@ public class Wrath : IPC
 {
     public override string Name => "WrathCombo";
     public override string Repo => Repos.Punish;
+    private static string InternalName => Svc.PluginInterface.InternalName;
```
Style wise I prefer variables used once to not be variables. Just put this directly in registerforlease

zbee replied:
Changed.

Jaksuhn commented on SomethingNeedDoing/External/Wrath.cs:
```diff
@@ -6,12 +6,15 @@ public class Wrath : IPC
-    public readonly Func<string, string, string, Guid?> RegisterForLeaseWithCallback = null!;
+        description: "Registers for lease",
+        parameterDescriptions: ["scriptName"])]
+    public Guid? RegisterForLease(string scriptName) => _registerForLease(InternalName, scriptName);
```
Direct ipcs should keep the exact name that they have in the other plugin. Any wrappers should get an alternative name

zbee replied:
Changed.

Jaksuhn commented on SomethingNeedDoing/External/Wrath.cs:
```diff
@@ -6,12 +6,15 @@ public class Wrath : IPC
-    public readonly Func<string, string, string, Guid?> RegisterForLeaseWithCallback = null!;
+        description: "Registers for lease",
+        parameterDescriptions: ["scriptName"])]
+    public Guid? RegisterForLease(string scriptName) => _registerForLease(InternalName, scriptName);
```
Please add the `Changelog` attribute for these changes. I've added a `ChangelogAttribute.Unreleased` you can pass as the version. It'll make adding much easier

zbee replied:
I believe I changed it as requested:

https://github.com/zbee/SND/blob/573d18dbfa4c1343b48e4b56f4c2916f77e3d608/SomethingNeedDoing/External/Wrath.cs#L15

zbee committed:
Merge remote-tracking branch 'upstream/main'

zbee committed:
fix(Wrath): Requested Changes

Changed the `RegisterForLease` alias to `Register`, and added a `Changelog` to it.
